automatically polyfill window.fetch with XMLHttpRequest (#7597)
cypress-io/cypress: packages/driver/README.md

localStorage.debug = "cypress:driver"
```

## Patches

- `sinon`

> - `unfetch` to polyfill `fetch`. Added `this` and changed function to arrow function to be able to load `fetch` in the spec window, but point the polyfill at the `window.XMLHttpRequest` in the application under test window.

Reviewer: I don't believe using `this.XMLHttpRequest` is correct because then the `fetch` function will not behave as it does natively based on the way it is called.

If you bind to `this.XMLHttpRequest` then calling the `fetch` function with a different context will fail. For example: `fetch.call({}, ...)` will fail. Likely `fetch.call(undefined, ...)` or `fetch.call(null, ...)` will fail depending on whether or not `use strict` is enabled.

This will natively happen if you use a splat in the app code such as `fetch(...args)` and the code is transpiled down to ES5. 

A better approach here is to **never** rely on the use of `this` and instead, update the `patch-package` to export a factory function that takes `contentWindow` as an argument, and then that returns you the `fetch` function. You would then change the `this.XMLHttpRequest` reference to use `contentWindow.XMLHttpRequest`, and no matter how `fetch` is called in the app code it will be bound correctly.

Example:

```js
export default function FetchFactory (contentWindow) => {
  return function fetch (...) => {
     // do the thing
     new contentWindow.XMLHttpRequest(...)
  }
}
```

Author: kk

> - `unfetch` to polyfill `fetch`. Added `this` and changed function to arrow function to be able to load `fetch` in the spec window, but point the polyfill at the `window.XMLHttpRequest` in the application under test window.

Author: ```suggestion
- `unfetch` to polyfill `fetch`. Added constructor function to point XMLHttpRequest to the application under test window.
```
